### PR TITLE
fix(internal_logs source): apply correct log level filtering

### PR DESCRIPTION
**Pull request by tobz**
This PR is a fix for #12251.

When previously switching over to per-layer filtering in `src/trace.rs` to better support using `tokio-console` without spamming the normal console logs with TRACE-level spans/events, we introduced a regression with internal logging.

For anyone using the `internal_logs` source, they would have started seeing TRACE-level logging flowing out of the source, even if the Vector logging level was at INFO.  We've fixed this by bringing `BroadcastSubscriber` -- the component responsible for feeding internal log events to `internal_logs` -- into the per-layer filtering fold.  This allows it to correctly filter events to match what is logged to console.

Fixes: #12251

### CODE REVIEW

File: src/trace.rs [hunk 1]
```diff
@@ -71,120 +91,150 @@ pub fn init(color: bool, json: bool, levels: &str) {
...
+    get_trace_sender().subscribe()
 }
 
-struct BroadcastSubscriber<S> {
-    subscriber: S,
+fn get_trace_subscriber_list() -> MutexGuard<'static, Option<Vec<oneshot::Sender<Vec<LogEvent>>>>> {
+    SUBSCRIBERS
+        .get_or_init(|| Mutex::new(Some(Vec::new())))
+        .lock()
+        .expect("poisoned locks are dumb")
```
Comment (jszwedko) [inline src/trace.rs:142]: Checks out.

**Review by jszwedko — COMMENTED** [review]
The changes seem reasonable to me, but cc/ing @bruceg to take a look since I think he wrote the initial implementation.

File: src/trace.rs (hunk 1 above)
Comment (tobz) [reply src/trace.rs:142]: I was surprised we didn't already pull in `parking_lot` as a dependency, which doesn't have poisonable locks... so this was mostly me just getting the boilerplate out of the way, but I would probably switch to `parking_lot` if this PR looked good overall.

**Review by bruceg — APPROVED** [review]

**Review by jszwedko — APPROVED** [review]
Awesome, thanks for splitting this PR up.

**Comment by jszwedko** [comment]
Cherry-picked as b49fe27e7

**Pull request merged**